iphone/Classes/TiViewProxy.m

```diff
@@ -230,27 +230,29 @@ -(void)add:(id)arg
 		pthread_rwlock_unlock(&childrenLock);
         
 		[childView setParent:self];
-
-		// Turn on clipping because I have children
-		[[self view] updateClipping];
-
-		[self contentsWillChange];
-		if(parentVisible && !hidden)
-		{
-			[childView parentWillShow];
-		}
+        
+        if (windowOpened)
+        {
+            // Turn on clipping because I have children
+            [[self view] updateClipping];
+            [self contentsWillChange];
+            if(parentVisible && !hidden)
+            {
+                [childView parentWillShow];
+            }
```

Comment:
Very interesting fix, will do a detailed review. But please fix the indentation already and try to use `if () {` instead, thx!